internal/store/model/device.go

```diff
@@ -167,12 +169,106 @@ func (dl DeviceList) ToApiResource(cont *string, numRemaining *int64) api.Device
 			ApplicationStatus: applicationStatuses,
 			SummaryStatus:     summaryStatuses,
 			UpdateStatus:      updateStatuses,
-			Total:             int64(len(dl)),
+			Total:             int64(len(*dl)),
 		},
 	}
 	if cont != nil {
 		ret.Metadata.Continue = cont
 		ret.Metadata.RemainingItemCount = numRemaining
 	}
-	return ret
+	return ret, nil
+}
+
+func DevicePtrReturnSelf(d *Device) *Device {
+	return d
+}
+
+func (d *Device) GetKind() string {
+	return api.DeviceKind
+}
+
+func (d *Device) GetName() string {
+	return d.Name
+}
+
+func (d *Device) GetOrgID() uuid.UUID {
+	return d.OrgID
+}
+
+func (d *Device) SetOrgID(orgId uuid.UUID) {
+	d.OrgID = orgId
+}
+
+func (d *Device) GetResourceVersion() *int64 {
+	return d.ResourceVersion
+}
+
+func (d *Device) SetResourceVersion(version *int64) {
+	d.ResourceVersion = version
+}
+
+func (d *Device) GetGeneration() *int64 {
+	return d.Generation
+}
+
+func (d *Device) SetGeneration(generation *int64) {
+	d.Generation = generation
+}
+
+func (d *Device) GetOwner() *string {
+	return d.Owner
+}
+
+func (d *Device) SetOwner(owner *string) {
+	d.Owner = owner
+}
+
+func (d *Device) GetLabels() JSONMap[string, string] {
+	return d.Labels
+}
+
+func (d *Device) SetLabels(labels JSONMap[string, string]) {
+	d.Labels = labels
+}
+
+func (d *Device) GetAnnotations() JSONMap[string, string] {
+	return d.Annotations
+}
+
+func (d *Device) SetAnnotations(annotations JSONMap[string, string]) {
+	d.Annotations = annotations
```

Comment:
Rather than repeating all these for every resource, would it make sense to compose the resources from a common ObjectMeta and provide a generic accessor method for that? And if so, would we still need getters/setters?

Reply:
All these except `GetKind` are part of `Resource`.  So all these can be implemented in `Resource` and remove from all others.

Reply:
I removed it all